pkg/env/env.go

```diff
@@ -313,8 +321,8 @@ func (e *testEnv) TestInParallel(t *testing.T, testFeatures ...types.Feature) {
 //
 // BeforeTest and AfterTest operations are executed before and after
 // the feature is tested respectively.
-func (e *testEnv) Test(t *testing.T, testFeatures ...types.Feature) {
-	e.processTests(t, false, testFeatures...)
+func (e *testEnv) Test(t *testing.T, testFeatures ...types.Feature) context.Context {
```

Comment:
You are right, we could actually omit it, I added it because I felt it could be useful for users to inspect the content of the context after calling it, but now that I think about it, it's more an implementation detail of testenv rather than of the Environment

Reply:
@phisco funny thing is while researching the race condition issue, i ran into the need to inspect the context. However, I would lean toward not leaving the signature as is to avoid breakage.  

Maybe a separate PR would be useful to update the signature of the Test and TestInParallel methods. That way, it's effect can be studied (or roll back if needed) with minimal disruption.

Reply:
@vladimirvivien, I did some checks, adding a returned context to the signatures would not be breaking change, maybe some linter could complain, but it will build without any issue. However, not returning it, would mean there is no way to inspect the content of the ctx after a run as we were saying, which would mean that we would have no way to  check the following test cases:
- https://github.com/kubernetes-sigs/e2e-framework/pull/292/files#diff-6da391e4e882fa4bb01f054410d115230af7831328b51bcf92939c9bfae905aaL318-R320 already existing
- https://github.com/kubernetes-sigs/e2e-framework/pull/292/files#diff-6da391e4e882fa4bb01f054410d115230af7831328b51bcf92939c9bfae905aaR374-R375 which I added, similar to the existing one
- https://github.com/kubernetes-sigs/e2e-framework/pull/292/files#diff-6da391e4e882fa4bb01f054410d115230af7831328b51bcf92939c9bfae905aaR424-R425 which I added, similar to the existing one

So, I tried adding two internal `testEnv.test/testInParallel` functions that are returning the ctx, which are then called directly by their public versions `Test/TestInParalle`, explicitly discarding the context returned, so that we can use those in tests if we need to check the ctx afterward, but they won't be accessible to users.

I would probably suggest adding the return ctx to the interface, but I totally understand not wanting to change the public API at least as part of this PR, so this could be a good intermediate step while we take a decision around that.